openprescribing/measure_definitions/cgm_sensors.json

```diff
@@ -0,0 +1,41 @@
+{
+  "name": "Prescribing of continuous glucose monitoring sensors",
+  "title": "",
+  "description": "Prescribing of continuous glucose monitoring sensors per 1000 patients",
+  "numerator_short": "Continuous glucose monitoring sensor quantity",
+  "denominator_short": "1000 patients",
+  "why_it_matters": [
+    "Continuous glucose monitoring devices (CGM) can significantly reduce the need for people with diabetes to conduct finger-prick testing.</p>",
+    "<p>NICE guidance for <a href='https://www.nice.org.uk/guidance/ng17'>Type 1 diabetes in adults</a>, <a href='https://www.nice.org.uk/guidance/ng28'>",
+    "Type 2 diabetes in adults</a> and <a href='https://www.nice.org.uk/guidance/ng18'>Type 1 and Type 2 diabetes in children and young people</a> were all ",
+    "updated in March 2022. The updates include recommendations that all patients with type 1 diabetes should be offered a choice of real-time continuous ",
+    "glucose monitoring (rtCGM) or intermittently scanned continuous glucose monitoring (isCGM) and that adults with type 2 diabetes on multiple daily ",
+    "insulin injections should be offered GM where they meet specific clinical criteria."
+  ],
+  "tags": [
+    "core",
+    "diabetes"
+  ],
+  "url": null,
+  "is_percentage": false,
+  "is_cost_based": false,
+  "low_is_good": null,
+  "numerator_type": "custom",
+  "numerator_columns": "SUM(p.quantity) AS numerator",
+  "numerator_from": [
+    "{hscic}.normalised_prescribing p -- joins prescribing data to subquery below, filtering list to BNF codes created in subquery \n",
+    "INNER JOIN -- joins the prescribing data table to the subquery thereby filtering the prescribing data to only these codes \n",
+    "(SELECT DISTINCT COALESCE (amp.bnf_code, vmp.bnf_code) bnf_code -- this creates a subquery to create a list of BNF codes from the VMP codes described below \n",
+    "FROM {dmd}.vmp vmp -- COALESCE function includes BNF codes from both the AMP and VMP tables (branded and generic), and DISTINCT means they're not duplicated \n",
+    "INNER JOIN {dmd}.amp amp ON vmp.id = amp.vmp -- joins AMP and VMP tables together on VMP code, allowing for BNF codes for branded and generic \n",
+    "WHERE vmp.id = 34865511000001109) bnf -- VMP for Glucose interstitial fluid detection sensor \n",
+    "ON p.bnf_code = bnf.bnf_code"
```

Comment:
This inner `SELECT` works in this case, but won't in general -- for instance, if all AMPs have a BNF code that doesn't match the VMP's BNF code, then the VMP's BNF code will be missed out.

I'd suggest replacing it with:

```
SELECT bnf_code FROM dmd.vmp
WHERE id = 34865511000001109

UNION DISTINCT

SELECT bnf_code FROM dmd.amp
WHERE vmp = 34865511000001109 AND bnf_code IS NOT NULL
```

Reply:
changed to suggested code, preview has run and is all good.  Thanks @inglesp